script/scp-hosts.sh

```diff
@@ -47,11 +42,17 @@ do
   echo "scp dirs to $host/$installPath starting"
 	ssh -p $sshPort $host  "cd $installPath/; rm -rf bin/ conf/ lib/ script/ sql/ ui/"
 
+  for i in ${!workerNames[@]}; do
+    if [[ ${workerNames[$i]} == $host ]]; then
+      workerIndex=$i
+      break
+    fi
+  done
+
   for dsDir in bin master-server worker-server alert-server api-server ui
   do
-    # if worker in workersGroupMap
-    if [[ "${workersGroupMap[${host}]}" ]]; then
-      echo "export WORKER_GROUPS_0=${workersGroupMap[${host}]}" >> worker-server/bin/dolphinscheduler_env.sh
+    if [[ ! -z ${workerIndex+x} ]]; then
+      echo "export WORKER_GROUPS_0=${groupNames[$workerIndex]}" >> worker-server/bin/dolphinscheduler_env.sh
```

Comment:
This path has change to ```worker-server/conf/application.yaml```. And the variables ```WORKER_GROUPS_0``` is no longer exists.
https://github.com/apache/dolphinscheduler/blob/a6c14fb23369ae37a1f5f0232ace90fa7a39435f/dolphinscheduler-worker/src/main/resources/application.yaml#L72-L73

Reply:
So can I simply remove line 53 and 54 as it has already been defined in each worker's application.yaml?

Reply:
No. we should adapt this logic to ```application.yaml```